packages/supabase_flutter/README.md

```diff
@@ -427,8 +427,167 @@ Supabase.initialize(
+        if (!kIsWeb && boxPath != null) {
+          final boxDir = File(boxPath).parent;
+          final dirIsEmpty = await boxDir.list().length == 0;
+          if (dirIsEmpty) {
+            await boxDir.delete();
+          }
+        }
+      }
+    }
+  }
+
+  @override
+  Future<String?> accessToken() {
+    return sharedPreferencesLocalStorage.accessToken();
+  }
+
+  @override
+  Future<bool> hasAccessToken() {
+    return sharedPreferencesLocalStorage.hasAccessToken();
+  }
+
+  @override
+  Future<void> persistSession(String persistSessionString) {
+    return sharedPreferencesLocalStorage.persistSession(persistSessionString);
+  }
+
+  @override
+  Future<void> removePersistedSession() {
+    return sharedPreferencesLocalStorage.removePersistedSession();
+  }
+}
+
+/// A [LocalStorage] implementation that implements Hive as the
+/// storage method.
+class HiveLocalStorage extends LocalStorage {
+  /// Creates a LocalStorage instance that implements the Hive Database
+  const HiveLocalStorage();
+
+  /// The encryption key used by Hive. If null, the box is not encrypted
+  ///
+  /// This value should not be redefined in runtime, otherwise the user may
+  /// not be fetched correctly
+  ///
+  /// See also:
+  ///
+  ///   * <https://docs.hivedb.dev/#/advanced/encrypted_box?id=encrypted-box>
+  static String? encryptionKey;
+
+  @override
+  Future<void> initialize() async {
+    HiveCipher? encryptionCipher;
+    if (encryptionKey != null) {
+      encryptionCipher = HiveAesCipher(base64Url.decode(encryptionKey!));
+    }
+    await Hive.initFlutter('auth');
+    await Hive.openBox(_hiveBoxName, encryptionCipher: encryptionCipher)
+        .timeout(const Duration(seconds: 1));
+  }
+
+  @override
+  Future<bool> hasAccessToken() {
+    return Future.value(
+      Hive.box(_hiveBoxName).containsKey(
+        supabasePersistSessionKey,
+      ),
+    );
+  }
+
+  @override
+  Future<String?> accessToken() {
+    return Future.value(
+      Hive.box(_hiveBoxName).get(supabasePersistSessionKey) as String?,
+    );
+  }
+
+  @override
+  Future<void> removePersistedSession() {
+    return Hive.box(_hiveBoxName).delete(supabasePersistSessionKey);
+  }
+
+  @override
+  Future<void> persistSession(String persistSessionString) {
+    // Flush after X amount of writes
+    return Hive.box(_hiveBoxName)
+        .put(supabasePersistSessionKey, persistSessionString);
+  }
+}
+```
+
+You can then initialize Supabase with `MigrationLocalStorage` and it will automatically migrate the session from Hive to SharedPreferences.
+
+```dart
+Supabase.initialize(
+  // ...
+  authOptions: FlutterAuthClientOptions(
+    localStorage: const MigrationLocalStorage(
+      persistSessionKey:
+              "sb-${Uri.parse(url).host.split(".").first}-auth-token",
+    ),
+  ),
+);
+```
+
 ---
 
+## Migrating Guide
+
+You can find the migration guide to migrate from v1 to v2 here:
+https://supabase.com/docs/reference/dart/upgrade-guide
+
```

Comment:
@Vinzent03 
I've added the migration guide here, but not sure if this is the ideal spot. Open to suggestions!

Reply:
I would additionally add it [here](https://github.com/supabase/supabase-flutter/blob/main/packages/supabase_flutter/CHANGELOG.md#200), because that's what I would mostly look for.

Reply:
Will add it there when publishing the package!